x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -462,4 +482,36 @@ public void testValidateForFipsNoErrorsForDefaultSettings() {
         Security.validateForFips(settings);
         // no exception thrown
     }
+
+    public void testLicenseUpdateFailureHandlerUpdate() throws Exception {
+        Settings settings = Settings.builder().
+            put("xpack.security.authc.api_key.enabled", "true").
+            build();
+        Collection<Object> components = createComponentsWithSecurityNotExplicitlyEnabled(settings);
+        AuthenticationService service = findComponent(AuthenticationService.class, components);
+        assertNotNull(service);
+        RestRequest request = new FakeRestRequest();
+        final AtomicBoolean completed = new AtomicBoolean(false);
+        service.authenticate(request, ActionListener.wrap(result -> {assertTrue(completed.compareAndSet(false, true));
+                                                                    }, this::logAndFail));
```

Comment:
Our typical style is:

```suggestion
        service.authenticate(request, ActionListener.wrap(result -> {
            assertTrue(completed.compareAndSet(false, true));
        }, this::logAndFail));
```

Reply:
Changed